test/tests/serial/vmail-id.bats

```diff
@@ -41,35 +40,27 @@ function teardown_file() { _default_teardown ; }
 # This test case is shared with tests.bats, but provides context on errors + some minor edits
 # TODO: Could improve in future with keywords from https://github.com/docker-mailserver/docker-mailserver/pull/3550#issuecomment-1738509088
 # Potentially via a helper that allows an optional fixed number of errors to be present if they were intentional
-@test '/var/log/mail/mail.log is error free' {
+@test 'Mail log is error free' {
   # Postfix: https://serverfault.com/questions/934703/postfix-451-4-3-0-temporary-lookup-failure
-  _run_in_container grep 'non-null host address bits in' /var/log/mail/mail.log
-  assert_failure
+  _service_log_should_not_contain_string 'mail' 'non-null host address bits in'
 
   # Postfix delivery failure: https://github.com/docker-mailserver/docker-mailserver/issues/230
-  _run_in_container grep 'mail system configuration error' /var/log/mail/mail.log
-  assert_failure
+  _service_log_should_not_contain_string 'mail' 'mail system configuration error'
 
   # Unknown error source: https://github.com/docker-mailserver/docker-mailserver/pull/85
-  _run_in_container grep -i ': error:' /var/log/mail/mail.log
-  assert_failure
+  _service_log_should_not_contain_string 'mail' ': error:'
 
   # Unknown error source: https://github.com/docker-mailserver/docker-mailserver/pull/320
-  _run_in_container grep -i 'not writable' /var/log/mail/mail.log
-  assert_failure
-  _run_in_container grep -i 'permission denied' /var/log/mail/mail.log
-  assert_failure
+  _service_log_should_not_contain_string 'mail' 'not writable'
+  _service_log_should_not_contain_string 'mail' 'permission denied'
 
   # Amavis: https://forum.howtoforge.com/threads/postfix-smtp-error-caused-by-clamav-cant-connect-to-a-unix-socket-var-run-clamav-clamd-ctl.81002/
-  _run_in_container grep -i '(!)connect' /var/log/mail/mail.log
-  assert_failure
+  _service_log_should_not_contain_string 'mail' '\(!\)connect'
```

Comment:
This one is correct and doesn't seem like it needs any different casing (_as most probably don't with `-i`_).

It'd be good if this was not catering to regex 😅

Reply:
> This one is correct and doesn't seem like it needs any different casing (_as most probably don't with `-i`_).

Since I added `-i` to the `grep` statement in `_filter_service_log`, this PR introduces the maybe unwanted situation in which the casing does not matter at all. I'll play a bit around with that.

> It'd be good if this was not catering to regex 😅

Yes, but it was easier to implement it first for a regex. I will also play around with that.

Reply:
Partially implemented in aaeea8af8100fa52e5087abe87b007726e7826de
Converted all usage of helpers in b3a6a7b103862d601f7b9bd0b1cc33d4a632e337